about/design-goals.md

```diff
@@ -0,0 +1,52 @@
+# CloudEvents Specification Design Goals
+
+CloudEvents are typically used in a distributed system to allow for services to
+be loosely coupled during development, then configured with tight-coupling for
+specific applications.
+
+The goal of the CloudEvents specification is to define interoperability of event
+systems that allow cloud providers, third-party tools and services to provide:
+- new event sources
+- new event consumers
+- observability, such as monitoring, debugging and visualization,
+- developer tools and libraries
+
+## Key properties of the system:
+- Source & action may be developed and deployed independently
+- Trigger configuration allows for actions to be invoked based on a condition
+(e.g. for a specific event type generated by a source, optionally filtered by
+specific attribute values, such as a Function which is invoked when an image is
+uploaded to specific S3 or Google Cloud Storage “bucket” path)
+- Events are uniquely identified to facilitate developer velocity and tooling:
+  - a unique idenfier allows for idempotency for reliable implementation
+  - structured event definitions allow the consumer to rely on a specific format
+- Common metadata attributes allow for observability via tools whithout
+knowledge of the specific implementation of an event system and also can provide
+value without detailed handling of the specifics of the unique event data
+- CloudEvents may be transported from source to action with different protocols.
+
+In order to achieve the above goals and desired system properties, the
+Serverless WG must agree on a common vocabulary with precise definitions
+describing the key elements of the system including:
+- [Source](../spec.md#source)
+- Action <use definition from starter doc>
+- Trigger: the condition that causes an event to be transmitted from source to
+action
+- Rule: the association of trigger + action
+
+In order to achieve the interoperability goals, the Serverless WG must describe:
+- One or more common architectures that are in active use today or planned to be
+built by WG members
+- How events are transported from source to action via at least one protocol
+- Whether a shared trigger specification is needed, or if runtime
+interoperability can be achieved independently, then shared configuration and/or
+API definition can be addressed as follow-on work
+- Authorization model(s) that control access and execution permissions
+
+
+## Non-Goals of this spec
+The following non-goals may be follow-on work, yet would not be included in the
+1.0 specification.
+- Function build and invocation process
+- Language-specific runtime APIs
```

Comment:
If we are achieving semantic and wire-level interoperability across systems, we have already achieved a lot. 

An API definition, or an API language binding, or a dispatch model for how to map event data to code would seem to define how a user application interacts with events on either the source side or the subscriber side, and that's even another layer further out into the app from "Trigger" and "Action" as laid out above. 

In the messaging space, we begrudgingly need to carry the "shared API" JMS legacy around with us because of the installed base and are doing mapping work in OASIS AMQP BINDMAP, but you could few of us excited about building another shared API standard. JMS severely constrains access to features, it imposes a particular notion of topology, it imposes a particular notion of send/receive patterns, it imposes a particular notion of filters, and so on. The innovation now all happens on different client stacks, on in some cases, like Apache Qpid Proton or AMQP for Net, those client stacks either work across products with their own API, or are the basis for product SDKs that wrap them. That all exists interoperably without the OASIS AMQP TC having imposed an API shape on those projects.

Reply:
@clemensv are you looking for a change or is this just a commentary?

Reply:
I am suggesting for qualification above to be dropped and clearly state that we're not defining APIs.

Reply:
@clemensv I'm inclined to agree, yet others have advocated for APIs as a project of the Serverless WG.  In any case, I think I can re-word so as to simply clarify it is outside of our current scope